packages/blocks/src/__internal__/rich-text/rich-text.ts

```diff
@@ -10,8 +10,13 @@ import { isValidUrl } from '../utils/url.js';
 import { setupVirgoScroll } from '../utils/virgo.js';
 import { createKeyboardBindings, createKeyDownHandler } from './keyboard.js';
 import { REFERENCE_NODE } from './reference-node.js';
-import { type AffineTextSchema, type AffineVEditor } from './virgo/types.js';
-const IGNORED_ATTRIBUTES = ['code', 'reference'] as const;
+import {
+  type AffineTextAttributes,
+  type AffineTextSchema,
+  type AffineVEditor,
+} from './virgo/types.js';
+const EDGE_IGNORED_ATTRIBUTES = ['code', 'reference'] as const;
+const GLOBAL_IGNORED_ATTRIBUTES = ['reference'] as const;
```

Comment:
Wondering why here requires two levels of ignore?

Reply:
It seems that it needs some comments and test, just hold on.

Reply:
In the beginning, this is a requirement proposed by a product manager.

When you input something after a link or code, the new text should not format as code or link. [PR](https://github.com/toeverything/blocksuite/pull/1822/files#diff-3a54b6d597a9c512f8d8484beffb57b71e7b89bb20085618f270116618f280b5R100-R101)

With the addition of the link automatic detection feature, the linked words are allowed to maintain the hyperlink format (with updated content). [PR](https://github.com/toeverything/blocksuite/pull/2624/files#diff-3a54b6d597a9c512f8d8484beffb57b71e7b89bb20085618f270116618f280b5R14)

After adding the linked page feature, it should not be possible for user input to extend the `reference` format.